hivemind/dht/protocol.py

```diff
@@ -221,9 +235,16 @@ async def rpc_store(self, request: dht_pb2.StoreRequest, context: grpc.ServicerC
             asyncio.create_task(self.rpc_ping(dht_pb2.PingRequest(peer=request.peer), context))
         assert len(request.keys) == len(request.values) == len(request.expiration_time) == len(request.in_cache)
         response = dht_pb2.StoreResponse(store_ok=[], peer=self.node_info)
-        keys = map(DHTID.from_bytes, request.keys)
-        for key_id, tag, value_bytes, expiration_time, in_cache in zip(
+        for key, tag, value_bytes, expiration_time, in_cache in zip(
                 keys, request.subkeys, request.values, request.expiration_time, request.in_cache):
+            if self.record_validator is not None:
+                try:
+                    self.record_validator.validate(ProtectedRecord(*record), signature)
```

Comment:
Note: RecordValidator.validate returns bool, but this line ignores the bool value.

Request: if not record_validator.validate(...), response.store_ok.append(False); continue

Reply:
As I wrote below, the `RSASignatureValidator` implementation actually returns None when validation is successful and throws the exception if it is not.

I'd like to keep this behavior, since there are many possible reasons for a validation failure (invalid public key, unsupported signature format, incorrect signature) and it would be useful to know the exact reason for future debugging (we can just return it in code or even log a warning with it).

Unfortunately, I forgot to update `RecordValidatorBase.validate` signature to represent this logic, so I'll fix its signature.

Reply:
Done